internal/osbuild2/rpm_stage.go

```diff
@@ -70,10 +80,38 @@ type RPMStageReference struct {
 //    }
 // }
 // ...
-type RPMStageReferences map[string]*RPMStageReference
+type RPMStageReferences map[string]*RPMStageSourceOptions
```

Comment:
nitpick: IMO it would make sense to rename `RPMStageReferences` to something like `RPMStageSourceRefs` or even `RPMStageSourceMapRefs`, similar to `RPMStageSourceArrayRefs`.

Reply:
Similar to my other comment, yes it would be good to rename a lot of these to make more sense, and even define them elsewhere.  I considered removing this type altogether because it's not used at all now.  I kept it around so that we can clean it up in a followup redesign.

Reply:
Fair enough... 😉